content/algorithms/expected_consensus/_index.md

```diff
@@ -28,115 +28,290 @@ The [Storage Power Consensus](storage_power_consensus) subsystem uses access to
 
-```text
-const maxChallengeTicketSize = 2^len(H)
+Note that there are variants of this attack in which the miner introduces a new sector to change which sector will be challenged.
+
+> What happens if a sector expired after `WinningPoStSectorSetLookback`?
+
+An expired sector will not be challenged during WindowPoSt (hence not penalized after its expiration). However, an edge case around the fact that `WinningPoStSectorSetLookback` is longer than finality is that due to the lookback, a miner can be challenged for an expired sector between `expirationEpoch` and `expirationEpoch + WinningPoStSectorSetLookback - 1`. Therefore, it is important that miners keep an expired sector for `WinningPoStSectorSetLookback` more epochs after expiration, or they will not be able to generate a WinningPoSt (and get the corresponding reward).
+
+Example:
+- At epoch `X`:
+  - Sector expires and miner deletes the sector.
+- At epoch `X+WinningPoStSectorSetLookback-1`:
+  - The expired sector gets selected for WinningPoSt
+  - The miner will not be able to generate the WinningPoSt and they will not win the sector.
+
+**Step 3:** Block creation and propagation
+
+If the above is successful, miners build a block and propagate it.
+
+### GenerateElectionProof
+
+`GenerateElectionProof` outputs the result of whether a miner won the block or not as well as the quality of the block mined.
+
+The "Block Quality" is an integer that is used for weight and block reward calculations. For example a block with quality "2" is equivalent as two blocks of quality "1".
+
+#### High level algorithm
+
+- Get the percentage of power at block `ElectionPowerTableLookback`
+  - Get the power of the miner at block `ElectionPowerTableLookback`
+  - Get the total network power at block `ElectionPowerTableLookback`
+- Get randomness for the current epoch using `GetRandomness`.
+- Generate a VRF and compute its hash
+  - The storage miner's `workerKey` is given as input in the VRF process
+- Compute block quality: The smaller the hash is, the higher the block quality will be
+  - Compute the probability distribution of winning `k` blocks (i.e. Poisson distribution see below for details)
+  - Let `h_n` be the normalised VRF, i.e. `h_n = vrfout/2^256`
+  - The probability of winning one block is `1-P[X=0]`, thus if `h_n` is less than `1-P[X=0]`, the miner wins at least one block.
+  - Similarly if `h_n` is less than `1-P[X=0]-P[X=1]` we have at least two blocks and so on.
+
+#### Explanations - Poisson Sortition
+
+Filecoin is building on the principle that a miner possessing X% of network power should win as many times as X miners with 1% of network power in the election algorithm.
+
+This is modelled via a Binomial distribution with parameter`p=ExpectedLeaderPerRound/TotalPower` and `n= MinerPower`.
+
+For ease of implementation, we use the the Poisson distribution with parameter `lambda = MinerPower*ExepectedLeader` to approximate this Binomial distribution.
```

Comment:
I wouldn't say Poisson is an approximation for ease of implementation. I initially thought Binomial(ExpectedLeadersPerRound, MinerPower/TotalPower) was the "correct" or ideal distribution, but after much discussion with @jzimmerman, I came to realize this: since every miner is rolling uncorrelated/independent dice, and we want to be fully invariant to miner pooling (not just in expectation), Poisson is actually the ideal distribution.

The table below might still be worth presenting, but the framing should be something like "you might think binomial is better, but it isn't, but look, it doesn't matter much anyway" rather than "our approximation is good enough".